mp_cnn/main.py

```diff
@@ -50,35 +51,38 @@
     parser.add_argument('--tensorboard', action='store_true', default=False, help='use TensorBoard to visualize training (default: false)')
     parser.add_argument('--run-label', type=str, help='label to describe run')
     args = parser.parse_args()
-    args.cuda = not args.no_cuda and torch.cuda.is_available()
 
+    random.seed(args.seed)
     np.random.seed(args.seed)
     torch.manual_seed(args.seed)
-    if args.cuda:
+    if args.device != -1:
         torch.cuda.manual_seed(args.seed)
 
-    train_loader, test_loader, dev_loader = MPCNNDatasetFactory.get_dataset(args.dataset, args.word_vectors_file, args.batch_size, args.cuda, args.sample)
+    dataset_cls, embedding, train_loader, test_loader, dev_loader \
+        = MPCNNDatasetFactory.get_dataset(args.dataset, args.word_vectors_dir, args.word_vectors_file, args.batch_size, args.device)
 
     filter_widths = list(range(1, args.max_window_size + 1)) + [np.inf]
-    input_channels = 300
-    model = MPCNN(input_channels, args.holistic_filters, args.per_dim_filters, filter_widths, args.hidden_units, train_loader.dataset.num_classes, args.dropout, args.sparse_features)
-    if args.cuda:
+    model = MPCNN(embedding, args.holistic_filters, args.per_dim_filters, filter_widths,
+                    args.hidden_units, dataset_cls.NUM_CLASSES, args.dropout, args.sparse_features)
+
+    with torch.cuda.device(args.device):
```

Comment:
Why not just condition of args.device ? If it equals -1, then we do not move model to gpu.

Reply:
Yeah your suggestion is the way to go. I was just concerned about too many `if` checks in `train.py` since code doesn't look clean